hls4ml/writer/vivado_accelerator_writer.py

```diff
@@ -311,11 +311,15 @@ def write_wrapper_test(self, model):
 
     def write_board_script(self, model):
         '''
-        Write the tcl scripts to create a Vivado IPI project for the VivadoAccelerator
+        Write the tcl scripts and kernel sources to create a Vivado IPI project for the VivadoAccelerator
         '''
         filedir = os.path.dirname(os.path.abspath(__file__))
         copyfile(os.path.join(filedir, self.vivado_accelerator_config.get_tcl_file_path()),
                  '{}/design.tcl'.format(model.config.get_output_dir()))
+        # Generic alveo board
+        if self.vivado_accelerator_config.get_board().startswith('alveo'):
+            copytree(os.path.join(filedir, self.vivado_accelerator_config.get_krnl_rtl_src_dir()),
+                 '{}/src'.format(model.config.get_output_dir()), dirs_exist_ok=True)
```

Comment:
This `dirs_exist_ok` option was introduced in Python 3.8, but hls4ml has the minimum Python version 3.7 ([here](https://github.com/fastmachinelearning/hls4ml/blob/master/setup.cfg#L5)). It would be okay to increase that to 3.8.